docker.go

```diff
@@ -42,72 +42,123 @@ func generatePort(portStr string) (nat.Port, error) {
 	return port, nil
 }
 
-// startContainer starts a docker container and returns the container ID
-// as well as a websocket connection to the attach endpoint.
-func (h *handler) startContainer(ctrInfo containerInfo) (string, *websocket.Conn, error) {
-	image := ctrInfo.dockerImage
-	// Use default docker image when user does not provide any
-	if image == "" {
-		image = defaultDockerImage
+type containerOptions func(cfg *container.Config)
+
+func withPort(port nat.Port) containerOptions {
+	return func(ctrCfg *container.Config) {
+		if port == "" {
+			return
+		}
+		ctrCfg.ExposedPorts = nat.PortSet{
+			port: struct{}{},
+		}
+	}
+}
+
+func withDockerImage(image string) containerOptions {
+	return func(cfg *container.Config) {
+		cfg.Image = image
+		if image == "" {
+			// Use default docker image when user does not provide any
+			cfg.Image = defaultDockerImage
+		}
 	}
+}
 
-	// pull container image if we don't already have it
-	if err := h.pullImage(image); err != nil {
-		return "", nil, fmt.Errorf("pulling %s failed: %v", image, err)
+type hostOptions func(cfg *container.HostConfig)
+
+func withExposedPort(port nat.Port) hostOptions {
+	return func(cfg *container.HostConfig) {
+		if port == "" {
+			return
+		}
+		cfg.PortBindings = map[nat.Port][]nat.PortBinding{
+			port: []nat.PortBinding{
+				{
+					HostIP:   "0.0.0.0",
+					HostPort: string(port),
+				},
+			},
+		}
+	}
+}
+
+func withSecurityOptions(seccompProfile string) hostOptions {
+	return func(cfg *container.HostConfig) {
+		b := bytes.NewBuffer(nil)
+		if err := json.Compact(b, []byte(seccompProfile)); err != nil {
+			// this should be caught while development itself and not during runtime
+			panic(fmt.Sprintf("compacting json for seccomp profile failed: %v", err))
+		}
+		cfg.SecurityOpt = []string{
+			"no-new-privileges",
+			fmt.Sprintf("seccomp=%s", b.Bytes()),
+		}
+	}
+}
+
+func NewContainerConfig(opts ...containerOptions) *container.Config {
+	cfg := &container.Config{
+		Cmd:          []string{"sh"},
+		Tty:          true,
+		AttachStdin:  true,
+		AttachStdout: true,
+		AttachStderr: true,
+		OpenStdin:    true,
+		StdinOnce:    true,
 	}
 
-	securityOpts := []string{
-		"no-new-privileges",
+	for _, opt := range opts {
+		opt(cfg)
+	}
+
+	return cfg
+}
+
+func NewContainerHostConfig(opts ...hostOptions) *container.HostConfig {
+	cfg := &container.HostConfig{
+		CapDrop:     strslice.StrSlice{"NET_RAW"},
```

Comment:
I am not sure why it was removed. NET_RAW is part of Docker's default capabilities and it should be fine to give to the container, as long as we don't start it on the host network namespace with `--net=host`.

Reply:
Do you want me to remove this option?

Reply:
Yes, let's remove it.

Reply:
LGTM after that

Reply:
done